prediction_market_agent/agents/microchain_agent/call_api.py

```
@@ -0,0 +1,50 @@
+        """
+        return (
+            "We can use this function to make arbitrary API calls to any endpoint with any parameters. "
+            "It supports different HTTP methods (GET, POST, etc.) and can include headers, "
+            "query parameters, and request bodies."
+        )
+
+    @property
+    def example_args(self) -> list[str]:
+        """
+        Returns a list of strings representing the example arguments for the API call.
+
+        :return: A list of strings representing the example arguments for the API call.
+        :rtype: list[str]
+        """
+        return ["method", "url", "params", "data", "headers"]
+
+    def __call__(self, method: str, url: str, params: dict = None, data: dict = None, headers: dict = None) -> str:
+        """
+        Sends an HTTP request using the specified method to the given URL with optional parameters, data, and headers.
+
+        Args:
+            method (str): The HTTP method to use for the request.
+            url (str): The URL to send the request to.
+            params (dict, optional): Query parameters to include in the request. Defaults to None.
+            data (dict, optional): Request body to include in the request. Defaults to None.
+            headers (dict, optional): Headers to include in the request. Defaults to None.
+
+        Returns:
+            str: The response string if the request was successful, or the error message if an exception occurred.
+        """
+        try:
+            response = requests.request(method, url, params=params, data=data, headers=headers)
+            response.raise_for_status()
+            return response.text
+        except requests.exceptions.RequestException as e:
+            return {"success": False, "error": str(e)}
```

Could you remove the try-except block please, other tools will throw an exception as well, so to not fail silently.